framework/include/base/Moose.h

```diff
@@ -279,3 +279,12 @@ class ScopedCommSwapper
 #endif
 
 } // namespace Moose
+
+// TODO namespace this inside MOOSE as Moose::map_find once libmesh namespace is separate
+// This macro is useful to get around the deficiencies of the C++ map 'at' and 'find' operators
+// at errors with an exception which requires catching to output nicely
+// find is a very verbose syntax
+// This should only be used in a class where name() and type() are defined. If not, rely on
+// libmesh_map_find instead
+#define moose_map_find(map, key)                                                                   \
+  MooseUtils::map_find((map), (key), name(), type(), __FILE__, __LINE__)
```

Comment:
We can probably do that not as a macro but as a templated function with SFINAE   (passing in `this` - if you want you can wrap that in a macro). Then we can use it anywhere we like and get name/type if it's available, and fall back to vanilla libmesh_find if not.

Reply:
Alright gave it a go.

I cant wait to try C++20 for this, SFINAE is heinous to work with